Tests/Rules/UseShouldProcessForStateChangingFunctions.tests.ps1

```diff
@@ -42,5 +42,11 @@ Function New-{0} () {{ }}
         It "returns no violations" {
             $noViolations.Count | Should -Be 0
         }
+
+        It "Workflows should not trigger a warning because they do not allow SupportsShouldProcess" {
```

Comment:
I think we need to protect this from execution on core, where workflow is not supported. Add '-skip:$IsCore' I think.

Reply:
```
PS> Invoke-ScriptAnalyzer -ScriptDefinition "workflow foo { 'zap' }"                                                                        
Invoke-ScriptAnalyzer : Parse error in script definition:  Workflow is not supported in PowerShell Core at line 1 column 1.
At line:1 char:1
+ Invoke-ScriptAnalyzer -ScriptDefinition "workflow foo { 'zap' }"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : ParserError: (WorkflowNotSupportedInPowerShellCore:String) [Invoke-ScriptAnalyzer], ParseException
+ FullyQualifiedErrorId : Parse error in script definition:  Workflow is not supported in PowerShell Core at line 1 column 1.,Microsoft.Windows.PowerShell.ScriptAnalyzer.Commands.InvokeScriptAnalyzerCommand
```

Reply:
Good catch, thanks.